host_provider/providers/azure.py

```diff
@@ -0,0 +1,342 @@
+
+
+class AzureProvider(ProviderBase):
+    BasicInfo = namedtuple("AzureBasicInfo", "id")
+    azClient = None
+    connCls = AzureConnection
+
+    @classmethod
+    def get_provider(cls):
+        return Provider.AZURE_ARM
+
+    def build_client(self):
+        AzureClient = self.get_driver()
+
+        client = AzureClient(
+            self.credential.tenant_id,
+            self.credential.subscription_id,
+            self.credential.access_id,
+            self.credential.secret_key,
+            region=self.credential.region
+        )
+        return client
+
+    def build_credential(self):
+        return CredentialAzure(
+            self.get_provider(), self.environment, self.engine)
+
+    def get_azure_connection(self):
+        az = self.connCls()
+        self.azClient = az
+
+        return az.conn_cls(self.credential.access_id,
+                           self.credential.secret_key,
+                           tenant_id=self.credential.tenant_id,
+                           subscription_id=self.credential.subscription_id)
+
+    def get_node(self, node_id):
+        pass
+
+    def _parse_image(self, name, size, gallery='myGallery', image='mssql_2019_0_0', version='1.0.0'):
+        sql_dict = OrderedDict()
+        templates = JsonTemplates()
+        pw = self.credential.init_password
+        region = self.credential.region
+
+        image_id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s' \
+                   % (self.credential.subscription_id, self.credential.resource_group, gallery, image, version)
+
+        network_id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkInterfaces/%s' \
+                     % (self.credential.subscription_id, self.credential.resource_group, name)
+
+        osProfile = {'adminUsername': 'dbaas', 'computerName': name, 'adminPassword': pw}
+
+        try:
+            for file in templates.list_files(version):
+                file_name = file.name.split('.json')[0]
+                if file_name == 'sql':
+                    sql_dict = templates.load_json(file.as_posix())
+                    sql_dict['properties']['hardwareProfile']['vmSize'] = size
+                    sql_dict['properties']['storageProfile']['imageReference']['id'] = image_id
+                    sql_dict['properties']['storageProfile']['osDisk']['name'] = name
+                    sql_dict['properties']['osProfile'] = osProfile
+                    sql_dict['properties']['networkProfile']['networkInterfaces'][0]['id'] = network_id
+                    sql_dict['location'] = region
+            return sql_dict
+        except Exception as error:
+            raise Exception("Template parse error: {}" % error)
+
+    def offering_to(self, cpu, memory, api_version='2020-12-01'):
+
+        base_url = self.credential.endpoint
+        action = "subscriptions/%s/providers/Microsoft.Compute/locations/%s/vmSizes?api-version=%s" \
```

Comment:
- Hardcoded url.
- use format instead of %
- do no use \ to break lines, use parentesis